lib/init.coffee

```diff
@@ -1,8 +1,50 @@
+{BufferedProcess, CompositeDisposable} = require 'atom'
+
 module.exports =
+
   config:
     javaExecutablePath:
       type: 'string'
-      default: ''
+      title: 'Path to the javac executable'
+      default: '/usr/bin/javac'
 
   activate: ->
-    console.log 'activate linter-javac'
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.config.observe 'linter-javac.javaExecutablePath',
+      (newValue) =>
+        @javaExecutablePath = newValue
+
+  deactivate: ->
+    @subscriptions.dispose()
+
+  provideLinter: ->
+    grammarScopes: ['source.java']
+    scope: 'file'
+    lintOnFly: true
+    lint: (textEditor) =>
+      return new Promise (resolve, reject) =>
+        filePath = textEditor.getPath()
+        messages = []
+        process = new BufferedProcess
+          command: @javaExecutablePath
+          args: ['-Xlint:all', filePath]
+          stderr: (data) ->
+            # Regex to match the entire error/warning message including the caret (^)
+            # that points to the location of the issue in the source line.
+            regex = /java:(\d+): (error|warning): (.+)\r?\n.*\r?\n( *)\^/g
```

Comment:
Thanks!  I had considered using a block regex, to add comments and help make it more readable, but unfortunately, it doesn't look like we can use literal spaces within a block regex (https://github.com/jashkenas/coffeescript/issues/1199, https://github.com/jashkenas/coffeescript/issues/1668).  We can use `\x20`, but that makes it less readable.